wls-ergebnismeldung-service/src/test/java/de/muenchen/oss/wahllokalsystem/ergebnismeldungservice/rest/stimmzettelumschlaege/StimmzettelumschlaegeControllerIntegrationTest.java

```diff
@@ -0,0 +1,203 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.matching.UrlPattern;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.MicroServiceApplication;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.domain.stimmzettelumschlaege.Stimmzettelumschlaege;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.domain.stimmzettelumschlaege.StimmzettelumschlaegeRepository;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.exception.ExceptionConstants;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.service.stimmzettelumschlaege.StimmzettelumschlaegeModelMapper;
+import de.muenchen.oss.wahllokalsystem.wls.common.exception.rest.model.WlsExceptionCategory;
+import de.muenchen.oss.wahllokalsystem.wls.common.exception.rest.model.WlsExceptionDTO;
+import de.muenchen.oss.wahllokalsystem.wls.common.security.domain.BezirkUndWahlID;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import lombok.val;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+@SpringBootTest(classes = MicroServiceApplication.class)
+@AutoConfigureMockMvc
+@AutoConfigureWireMock
+@ActiveProfiles(
+        profiles = { SPRING_TEST_PROFILE, SPRING_NO_SECURITY_PROFILE, NO_BEZIRKS_ID_CHECK }
+)
+public class StimmzettelumschlaegeControllerIntegrationTest {
+
+    @Autowired
+    StimmzettelumschlaegeRepository stimmzettelumschlaegeRepository;
+
+    @Autowired
+    StimmzettelumschlaegeModelMapper stimmzettelumschlaegeModelMapper;
+
+    @Autowired
+    StimmzettelumschlaegeDTOMapper stimmzettelumschlaegeDTOMapper;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @AfterEach
+    void tearDown() {
+        stimmzettelumschlaegeRepository.deleteAll();
+    }
+
+    @Nested
+    class GetStimmzettelumschlaege {
+
+        @Test
+        void should_returnData_when_dataIsPresentInRepository() throws Exception {
+            val wahlID = "wahlID";
+            val wahlbezirkID = "wahlbezirkID";
+            val urneneroeffnungsUhrzeit = LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS);
```

Comment:
Würde ich eher in den Vergleich ziehen als in die Datenerstellung. Dafür gäbe es die Klasse `LocalDateTimeComparators` mit einen entsprechenden Comparator.

Würde auch spätere Stellen im Test betreffen.

Reply:
resolved with 2d1569a2b4ecaf5545dd812dbb126bfe89cdd035